example/README.md

```diff
@@ -4,39 +4,31 @@ This example demonstrates how to use the `sp1-solana` crate to verify a proof ge
 
 ## Overview: Script
 
-This script will load an SP1 [`SP1ProofWithPublicValues`]
-(https://docs.rs/sp1-sdk/2.0.0/sp1_sdk/proof/struct.SP1ProofWithPublicValues.html)
-from the pre-generated proof [`fibonacci_proof.bin`](./proofs/fibonacci_proof.bin). This is a SP1 Groth16 proof that
-proves that the 20th fibonacci number is 6765.
+1. Load an SP1 [`SP1ProofWithPublicValues`](https://docs.rs/sp1-sdk/2.0.0/sp1_sdk/proof/struct.SP1ProofWithPublicValues.html)
+from the pre-generated proof [`fibonacci_proof.bin`](../proofs/fibonacci_proof.bin). This is a SP1 Groth16 proof that
+proves that the 20th fibonacci number is 6765. Optionally, this proof can be freshly generated from
+the [`sp1-program`](../sp1-program).
 
-The loaded `SP1ProofWithPublicValues` will be serialized into a `SP1ProofFixture`, and written to
-[`proof-fixtures/fibonacci_fixture.bin`](./proof-fixtures/fibonacci_fixture.bin). The `SP1ProofFixture`
-can easily be borsh serialized and deserialized, and can be verified on chain.
+2. Serialize the `SP1ProofWithPublicValues` into a `SP1ProofFixture`, which is then written to
+[`proof-fixtures/fibonacci_fixture.bin`](../proof-fixtures/fibonacci_fixture.bin). Here is a snippet 
+from [`script/src/main.rs`](script/src/main.rs) that demonstrates this serialization.
 
-### Running the script
-
-To load the pregenerated proof and verify it, run the following commands. 
-
-```shell
-cd script
-RUST_LOG=info cargo run --release
-```
-
-To generate a fresh proof from the program in `sp1-program`, run the following commands. 
-
-```shell
-cd script
-RUST_LOG=info cargo run --release -- --prove
+```rust
+// Load the proof from the file, and convert it to a fixture.
+let sp1_proof_with_public_values = SP1ProofWithPublicValues::load(&proof_file).unwrap();
+let fixture = SP1ProofFixture::from(sp1_proof_with_public_values);
 ```
 
-## Overview: Solana Program
-
-The code in [`solana/program`](./solana/program) is a simple Solana contract that verifies a 
-`SP1ProofFixture` using the `sp1-solana` crate. It costs roughly 280,000 compute units. 
-
-Here is an excerpt from the Solana contract that demonstrates how to verify a `SP1ProofFixture` on chain. 
+3. Using the [`solana-program-test`](https://docs.rs/solana-program-test/latest/solana_program_test/) framework, send the `SP1ProofFixture` to the 
+[`fibonacci-verifier-contract`](./program). This smart contract will verify the proof using the `sp1-solana` crate,
+verify that the provided program vkey is correct, and print out the public inputs. Here is a snippet that demonstrates
+how to do some common operations on the SP1 proof fixture.
 
 ```rust
+// Derived by running `cargo prove vkey --elf ../../sp1-program/elf/riscv32im-succinct-zkvm-elf`.
```

Comment:
nit: I believe we're planning to deprecate this, we should suggest `vk.bytes32()` instead